### ISSUE TITLE
Lenovo S145

### ISSUE THREAD

**Issue by paulocoutinhox** [djale/Lenovo-Hackintosh#1]
Hi,

I have a **Lenovo S145 15IWL** and i have some questions :)

Im Paulo from other post on voodoo. I don't know if you remember me.

1 - Your wifi card mean that is not compatible with Lenovo:
https://pt.aliexpress.com/i/4000631751046.html

2 - Im already using EFI folder from Olarila:
https://www.olarila.com/topic/6874-olarila-hackbook-lenovo-ideapad-s145-mojave-catalina-full-dsdt-patches-clover-and-opencore/#comments

But i need now make trackpad and energy bar work. I extract my DSTS from Boot pressing F4:
https://www.dropbox.com/s/j72xa7nghfze20g/acpi-origin.zip?dl=1

What i need do now? Only need copy folder patched to my ACPI folder?

Can you help me :)

**Comment by djacicv** [comment]
1. It says for some reason it's not compatible with hp and Lenovo, I don't know why, but it's the same size as the one you already have, and antennas fit great with plastic cover. 
It works flawlessly, everything works on mac and windows, it has same chipset as original mac wifi card. I've got mine from <a href="https://pt.aliexpress.com/item/32464748097.html?spm=a2g0s.9042311.0.0.7004b90aDgzRvy">here</a>

2. You need to open your efi folder > clover > acpi > origin and copy DSDT.aml to folder "patched" after that restart.

3. The thing is, if you don't have everything patched before, this won't work.
Try using my EFI folder or at least copy patched lines from my config.plist and copy mine patched folder and replace it with yours.

4. Or you can find online how to patch your dsdt, there are tutorials.

Good luck :)

**Comment by paulocoutinhox** [comment]
I copy your folder EFI and put on EFI partition.

It show a screen to boot with a MI logo but always stop the progress bar.

![IMG_6309](https://user-images.githubusercontent.com/395096/83920766-4cc00c00-a753-11ea-8d78-74e760301505.png)

**Comment by djacicv** [comment]
Did you generated serial id and done the setup in config.plist before? You need to do that it's important.
Do you know how to do it, as I can't realize how much you are into this?

**Comment by paulocoutinhox** [comment]
No. My notebook already one EFI folder from Olarila. I deleted it and put yours to check. In your tutorial the serial is after enter on mac with your EFI, no?

**Comment by djacicv** [comment]
Yes, but you need to boot with USB with installation and proper EFI folder that works for you, to boot into the system, after that you need to copy new EFI folder and edit config.plist 

After booting to your system, with clover configurator, edit config.plist and generate serial id model number, etc.. (you can find a lot of tutorials on this). It's important to this properly because your facetime, iMessage and AppStore won't work.

**Comment by paulocoutinhox** [comment]
You are saying that im need reinstall? Because im already have a version with working app store etc.

**Comment by djacicv** [comment]
No, you don't need to reinstall the system, I said you need to have proper config.plist in order to get a working system with imessage and facetime and other stuff. 

When you boot with to your system with USB that you used to install system, do the stuff that I said, and give me feedback so we can go further.

**Comment by paulocoutinhox** [comment]
Is this that im not understanding. What USB? I don't use USB to boot.

**Comment by djacicv** [comment]
You posted few minutes ago that you can't boot to system, and posted picture? I said If you have trouble booting to system, use USB that you used to install macOS.

**Comment by paulocoutinhox** [comment]
The boot on system only NOT happen when i use your EFI folder. You mean that i need use your EFI folder but boot from USB?

**Comment by djacicv** [comment]
NO.

You need to read carefully, everything I post, take your time.
Can you boot back to your system, or not?

**Comment by paulocoutinhox** [comment]
Yes, and now i can boot using your EFI folder. But i need use USB to select "boot from macos".

The trackpad appear and the energy bar too.

**Comment by paulocoutinhox** [comment]
Im inside on macos system again using your EFI.

**Comment by djacicv** [comment]
Just to be clear.

1. Are you booting with USB or not?
2. Where did you put my efi folder, to system or USB?

**Comment by paulocoutinhox** [comment]
1 - Im booting from USB. If i don't boot from usb the notebook go to your logo/boot screen and when i select "boot from macos" it stop on that progressbar.

2 - I put your EFI into my SSD folder. The USB already have one.

**Comment by djacicv** [comment]
Okay, when you boot with USB it uses EFI that is on USB.

1. Open efi folder that is on your ssd and edit config.plist properly, after that try to boot again.
2. If that doesn't help you, put back EFI folder that you have on USB, and we will modify it to make it work.

**Comment by paulocoutinhox** [comment]
Ok. But what is the steps now. I already have the origin folder.

This are the steps now?

1 - I need copy DSTS.iml to patched folder?
2 - Run kextcache on terminal
3 - Reboot?

**Comment by djacicv** [comment]
1. Open efi folder that is on your ssd and edit config.plist properly, after that try to boot again.

That is the step, after that if you boot successfully, you need to do steps that you mentioned.

**Comment by paulocoutinhox** [comment]
But what you mean with "edit config properly"?

**Comment by djacicv** [comment]
<a href="https://youtu.be/IYismHReQ8A">See on this video</a> on 2:11 min he is setting up config.plist

**Comment by paulocoutinhox** [comment]
I followed the tutorial:
https://hackintosher.com/forums/thread/generate-your-own-hackintosh-serial-number-board-serial-number-uuid-mlb-rom-in-clover.306/

Generate new serial etc but the problem is the same. It stop on the progress bar.

**Comment by djacicv** [comment]
Boot in verbose and see what error message you get. 
When booting, go to boot options and add -v and boot

**Comment by paulocoutinhox** [comment]
Hi,

This is the log:
![IMG_6311](https://user-images.githubusercontent.com/395096/83934427-611dfc00-a787-11ea-8313-c28c4298f537.jpg)

**Comment by djacicv** [comment]
You need to check on tonymac how to properly do log dump. 
My suggestion is to do some more research about hackintosh overall and learn some basics.

Anyway, it's best to use your own EFI that works for you.
1.Use my kexts for trackpad (voodooi2c and Synaptics) - you need to remove your's voodoo kexts

After that I will help you to patch it, let me know how it goes.

**Comment by paulocoutinhox** [comment]
I replace mine kext folder with yours and now the trackpad stop work, only the mouse usb is working.

The energy bar appear and show energy normal.

I can't use my kext with your energy kext only? What i need do now?

The trackpad is unknown.

![image](https://user-images.githubusercontent.com/395096/83962598-0531a100-a875-11ea-95b4-985ea6b5edd7.png)

**Comment by djacicv** [comment]
It's important to read everything I say, as I can't repeat myself in each post. 

> 1.Use my kexts for trackpad (voodooi2c and Synaptics) - you need to remove your's voodoo kexts
> 
> After that I will help you to patch it, let me know how it goes.

**Comment by paulocoutinhox** [comment]
Hi,

Im already to what you say. And my trackpad not work. This is what happen.

In the boot log say that mine is ELAN. There is something wrong?

**Comment by paulocoutinhox** [comment]
And when i use only these two kexts that you sad the trackpad works, but other things like battery no and the trackpad still not found.
